core-bundle/src/EventListener/DataContainer/SubtitlesFieldsListener.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+use Contao\CoreBundle\File\SubtitlesType;
+use Contao\DataContainer;
+
+class SubtitlesFieldsListener
+{
+    #[AsCallback(table: 'tl_files', target: 'config.onpalette')]
+    public function addSubtitlesFields(string $palette, DataContainer $dc): string
+    {
+        // $dc->id is the file name in this case
+        if (str_ends_with($dc->id, '.vtt')) {
+            $palette = PaletteManipulator::create()
+                ->addField(['subtitlesLanguage', 'subtitlesType'], 'name')
+                ->applyToString($palette)
+            ;
+        }
+
+        return $palette;
+    }
+
+    #[AsCallback(table: 'tl_files', target: 'fields.subtitlesType.options')]
+    public function subtitlesTypeOptions(): array
+    {
+        return array_map(static fn ($case) => $case->name, SubtitlesType::cases());
+    }
```

Comment:
I would be fine with adding this one line inline to the DCA file as well - similar to what you did for getting the locales. 

IIRC we discussed that options could allow enums as a first class citizen in the future (so that you could also just specify the enum class and the driver would then extract the cases). I reckon this would make it easier to refactor, then. 

@leofeyer Wdyt?

Reply:
> I would be fine with adding this one line inline to the DCA file as well

Was thinking the same once I coded this, just waiting for more opinions here